### ISSUE TITLE
Using styled components in rules

### ISSUE THREAD

**Issue by modmanx** [solidjs/solid-styled-components#1]
Hello, I'm using solid-styled-components for a project. What I would like to do is to use component in rule like shown below as available in documentation https://github.com/cristianbote/goober#usage

```
<ComponentA>
  <ComponentB>
  </ComponentB>
</ComponentA>


export const ComponentB= styled('div')`
  width: 80px;
`;

export const ComponentA= styled('div')`
 &:hover ${ComponentB}{
    filter: brightness(1.5);
  }
`;
```

Unfortunately I'm getting error and do not know how to fix

```
solid.js:558 Uncaught (in promise) RangeError: Maximum call stack size exceeded
    at runComputation (solid.js:558)
    at updateComputation (solid.js:544)
    at createRenderEffect (solid.js:184)
    at spreadExpression (web.js:328)
    at spread (web.js:172)
    at index.js:49
    at goober.esm.js:1
    at Array.reduce (<anonymous>)
    at i (goober.esm.js:1)
    at Object.p (goober.esm.js:1)
```

Would anyone have any idea on how to proceed?

**Comment by ryansolid** [comment]
Thanks for reporting. I wasn't aware of this use case. I'm not using their styled-components directly, but also not doing much differently. I'll take a look.

**Comment by modmanx** [comment]
The best! Really appreciate your fast response.

**Comment by modmanx** [comment]
Do not worry if there isn't simple solution. Will then move to sass and use styled for other cases! Thank you for taking time.

**Comment by ryansolid** [comment]
I think I figured out a way to do this. Expose a helper on each Styled component specifically for this case. It is a bit of a weird one though. In that to get the className we calculate it based on the template and props. The props being passed in, in this case, are from the parent. I guess you could control your own props passed in.

**Comment by ryansolid** [comment]
Ok released in 0.26.4.
